src/s1reader/s1_burst_slc.py

```diff
@@ -263,17 +264,31 @@ def __str__(self):
     def __repr__(self):
         return f"{self.__class__.__name__}(burst_id={self.burst_id})"
 
-    def as_isce3_radargrid(self):
+    def as_isce3_radargrid(self, az_step: Optional[float] = None, rg_step: Optional[float] = None):
         '''Init and return isce3.product.RadarGridParameters.
 
-        Returns:
-        --------
+        Parameters
+        ----------
+        az_step : float, optional
+            Azimuth step size in seconds. If not provided, the azimuth step
+            size is set to the azimuth time interval.
+        rg_step : float, optional
+            Range step size in meters. If not provided, the range step size
+            is set to the range pixel spacing.
+
+        Returns
+        -------
         _ : RadarGridParameters
             RadarGridParameters constructed from class members.
         '''
 
+        if az_step is None:
+            az_step = self.azimuth_time_interval
         prf = 1 / self.azimuth_time_interval
```

Comment:
@scottstanie You need to also change the prf:

```suggestion
        prf = 1 / az_step
```